tests/test_parse.py

```diff
@@ -53,79 +53,174 @@ class MyModel(BaseModel):
             __root__: str
 
 
-@pytest.mark.xfail(reason='__root__ model')
 def test_model_validate_root():
-    # TODO: Migrate this test and those following to use root_validator, model_serializer, etc., once ready
-    #   Probably also want to re-add some of the __root__ tests from v1 that test other behaviors,
-    #   such as serialization and JSON schema.
     class MyModel(BaseModel):
-        __root__: str
-
+        root: str
+
+        # Note that the following three definitions require no changes across all __root__ models
+        # I couldn't see a nice way to create a decorator that reduces the boilerplate,
+        # but if we want to discourage this pattern, perhaps that's okay?
+        @root_validator(pre=True)
+        @classmethod
+        def populate_root(cls, values):
+            return {'root': values}
+
+        @model_serializer
+        def _serialize(self, info):
+            if info.mode == 'json':
+                return self.root
+            return self.__dict__
+
+        @classmethod
+        def model_modify_json_schema(cls, json_schema):
+            return json_schema['properties']['root']
+
+    # Validation
     m = MyModel.model_validate('a')
-    assert m.model_dump() == {'__root__': 'a'}
-    assert m.__root__ == 'a'
+    assert m.root == 'a'
+
+    # Serialization
+    # TODO: Possible concern — `model_dump` is annotated as returning dict[str, Any] — is that okay, given
+    #   model_serializer could change that? Should we try to reflect it in the mypy plugin?
+    assert m.model_dump() == {'root': 'a'}
+    assert m.model_dump_json() == '"a"'
```

Comment:
Adding this comment to draw attention to the TODO comment above

Reply:
Do you want to do that in this PR? I feel like that's may end up being unfortunate for ergonomics, but I guess safer than having a type error. And definitely better than making it generic over the dump type.

Reply:
let's leave it, users can always change the signature.